bridge/github/export.go

```diff
@@ -0,0 +1,731 @@
+	q := &labelQuery{}
+	variables := map[string]interface{}{
+		"label": githubv4.String(label),
+		"owner": githubv4.String(ge.conf[keyOwner]),
+		"name":  githubv4.String(ge.conf[keyProject]),
+	}
+
+	parentCtx := context.Background()
+	ctx, cancel := context.WithTimeout(parentCtx, defaultTimeout)
+	defer cancel()
+
+	if err := gc.Query(ctx, q, variables); err != nil {
+		return "", err
+	}
+
+	// if label id is empty, it means there is no such label in this Github repository
+	if q.Repository.Label.ID == "" {
+		return "", fmt.Errorf("label not found")
+	}
+
+	return q.Repository.Label.ID, nil
+}
+
+func (ge *githubExporter) createGithubLabel(label, color string) (string, error) {
+	url := fmt.Sprintf("%s/repos/%s/%s/labels", githubV3Url, ge.conf[keyOwner], ge.conf[keyProject])
+
+	client := &http.Client{
+		Timeout: defaultTimeout,
+	}
+
+	params := struct {
+		Name        string `json:"name"`
+		Color       string `json:"color"`
+		Description string `json:"description"`
+	}{
+		Name:  label,
+		Color: color,
+	}
+
+	data, err := json.Marshal(params)
+	if err != nil {
+		return "", err
+	}
+
+	req, err := http.NewRequest("POST", url, bytes.NewBuffer(data))
+	if err != nil {
+		return "", err
+	}
+
+	// need the token for private repositories
+	req.Header.Set("Authorization", fmt.Sprintf("token %s", ge.conf[keyToken]))
+
+	resp, err := client.Do(req)
+	if err != nil {
+		return "", err
+	}
+
+	if resp.StatusCode != http.StatusCreated {
+		return "", fmt.Errorf("error creating label: response status %v", resp.StatusCode)
+	}
+
+	aux := struct {
+		ID     int    `json:"id"`
+		NodeID string `json:"node_id"`
+		Color  string `json:"color"`
+	}{}
+
+	data, _ = ioutil.ReadAll(resp.Body)
+	defer resp.Body.Close()
+
+	err = json.Unmarshal(data, &aux)
+	if err != nil {
+		return "", err
+	}
+
+	return aux.NodeID, nil
+}
+
+// create github label using api v4
+func (ge *githubExporter) createGithubLabelV4(gc *githubv4.Client, label, labelColor string) (string, error) {
+	m := createLabelMutation{}
+	input := createLabelInput{
+		RepositoryID: ge.repositoryID,
+		Name:         githubv4.String(label),
+		Color:        githubv4.String(labelColor),
+	}
+
+	parentCtx := context.Background()
+	ctx, cancel := context.WithTimeout(parentCtx, defaultTimeout)
+	defer cancel()
+
+	if err := gc.Mutate(ctx, &m, input, nil); err != nil {
+		return "", err
+	}
+
+	return m.CreateLabel.Label.ID, nil
+}
+
+func (ge *githubExporter) getOrCreateGithubLabelID(gc *githubv4.Client, repositoryID string, label bug.Label) (string, error) {
+	// try to get label id
+	labelID, err := ge.getGithubLabelID(gc, string(label))
+	if err == nil {
+		return labelID, nil
+	}
+
+	// RGBA to hex color
+	rgba := label.RGBA()
+	hexColor := fmt.Sprintf("%.2x%.2x%.2x", rgba.R, rgba.G, rgba.B)
+
+	// create label and return id
+	// NOTE: since createLabel mutation is still in preview mode we use github api v4 to create labels
```

Comment:
```suggestion
	// NOTE: since createLabel mutation is still in preview mode we use github api v3 to create labels
```

Reply:
this function has nothing specific to V3 or V4, this comment should be elsewhere

Reply:
Moved to createLabel function